core/jobs.py

```diff
@@ -165,35 +163,52 @@ def register_completion(cls, job_id, output_list):
 
         model.status_code = STATUS_CODE_COMPLETED
         model.time_finished_msec = utils.get_current_time_in_millisecs()
+        model.output = cls._compress_output_list(output_list)
+        model.put()
+
+        cls._post_completed_hook(job_id)
 
-        # TODO(bhenning): Add tests for this.
-        output_str_list = ['%s' % output_value for output_value in output_list]
+    @classmethod
+    def _compress_output_list(
+            cls, output_list, test_only_max_output_chars=None):
+        """Returns compressed list of strings with a max length of chars.
+
+        Args:
+            output_list: list(*). Collection of objects to be stringified.
+            test_only_max_output_chars: int or None. Used to help test this
```

Comment:
I think there needs to be some description here of what this arg means, and what None means (so, not just the context in which it is used).